TombEngine/Game/collision/collide_item.cpp

```diff
@@ -828,6 +828,44 @@ bool ItemPushStatic(ItemInfo* item, const MESH_INFO& mesh, CollisionInfo* coll)
 	return true;
 }
 
+void CollideBridgeItems(CollisionResult& collResult, CollisionInfo& coll)
+{
+	// Store an offset for a bridge item into shifts, if exists.
+	if (coll.LastBridgeItemIndex == collResult.Position.Bridge && coll.LastBridgeItemIndex != NO_ITEM)
+	{
+		auto deltaPose = Pose(g_Level.Items[collResult.Position.Bridge].Pose.Position - coll.LastBridgeItemPose.Position,
+							  g_Level.Items[collResult.Position.Bridge].Pose.Orientation - coll.LastBridgeItemPose.Orientation);
+
+		auto heightDelta = coll.Setup.OldPosition.y - coll.LastBridgeItemPose.Position.y;
+
+		if (heightDelta >= 0 && (deltaPose.Position != Vector3i::Zero || deltaPose.Orientation != EulerAngles::Zero))
+		{
+			auto& bridgePos = g_Level.Items[collResult.Position.Bridge].Pose.Position;
+			auto distance = (coll.Setup.OldPosition - bridgePos).ToVector3();
+
+			float sin = phd_sin(deltaPose.Orientation.y);
+			float cos = phd_cos(deltaPose.Orientation.y);
+			auto x = distance.x * cos - distance.z * sin + bridgePos.x;
+			auto z = distance.x * sin + distance.z * cos + bridgePos.z;
+
+			deltaPose.Position.x += round((float)coll.Setup.OldPosition.x - x);
+			deltaPose.Position.z += round((float)coll.Setup.OldPosition.z - z);
+		   
+			// Don't update the shifts if difference is too big (possibly bridge was teleported or just entered bridge).
+			if (deltaPose.Position.ToVector3().Length() <= coll.Setup.Radius)
```

Comment:
Does this mean if the bridge's velocity is too big, the player won't go along with it anymore?

Reply:
Yes, this is done to imitate "inertia" with collision setup's radius field serving as a substitute for a mass.